### PR TITLE
Adding accessibility properties and support for automated testing (Appium like)

### PR DESCRIPTION
**Pull request by dsolimando**

### CODE REVIEW

File: src/NativePinInput.xml [hunk 1]
```diff
@@ -12,23 +18,24 @@
...
-                <description></description>
+                <description />
             </property>
         </propertyGroup>
         <propertyGroup caption="Styling">
             <property key="darkMode" type="enumeration" defaultValue="device">
                 <caption>Dark mode</caption>
-                <description>Choose whether the widget will follow the device dark mode setting or a fixed setting.</description>
+                <description
+                >Choose whether the widget will follow the device dark mode setting or a fixed setting.</description>
```
Comment (mgroeneweg) [inline src/NativePinInput.xml:38]: Why did this change happen? XML looks weird this way.

**Comment by mgroeneweg** [comment]
Thank you for the pull request!

However I missed the updated test app. You have changed and added widget properties so the test app in the repo needs changing as well.

Also, please make sure the widget still works properly on iOS and Android, smartphone and tablet.

Without updated test app I cannot accept the pull request.

I also noticed some weird changes in the property XML, please check the XML on your end and perhaps use the VS Code formatting option.

File: src/NativePinInput.xml (hunk 1 above)
Comment (dsolimando) [reply src/NativePinInput.xml:38]: typo :)

**Comment by dsolimando** [comment]
Hello,

I fixed your attention points.

Thanks

**Comment by mgroeneweg** [comment]
Hi,

Thanks! I am bringing my widgets forward to 9.24, I will include your pull requests then. Due to project deadlines, my widget work has to wait though.

File: src/components/PinInputButton.tsx [hunk 2]
```diff
@@ -22,5 +24,16 @@ export function PinInputButton({ caption, style, onClick }: PinInputButtonProps)
...
+            {...(!testID && {
+                accessibilityRole: "button",
+                accessibilityLabel: caption
+            })}
+            {...(testID && e2eID(testID))}
+            onPress={() => onClickHandler()}
+        >
+            {renderView}
+        </Pressable>
+    );
```
Comment (mgroeneweg) [inline src/components/PinInputButton.tsx:38]: I'm having a difficult time understanding what you want to achieve here. The e2eID function only returns accessibility props for Android. So, if the testID is set, only Android gets accessibility props, but if the testID is not set the accessibility props are set regardless of the platform?

File: src/components/DeleteButton.tsx [hunk 3]
```diff
@@ -19,11 +27,22 @@ export function DeleteButton({ deleteButtonIcon, style, onClick }: DeleteButtonP
...
     }, [deleteButtonIcon, style]);
 
-    return <Pressable onPress={() => onClick()}>{renderIcon}</Pressable>;
+    return (
+        <Pressable
+            {...(!testID && {
+                accessibilityRole: "button",
+                accessibilityLabel
+            })}
+            {...(!accessibilityLabel && { testID })}
```
Comment (mgroeneweg) [inline src/components/DeleteButton.tsx:42]: What does this code intend to achieve? I find this code difficult to understand

**Comment by mgroeneweg** [comment]
Hi,

I finally got around looking into this again. Please help me understand your changes, see above.

**Comment by mgroeneweg** [comment]
I have published a new version with accessibility support and test IDs

**Pull request closed without merging**